components/brave_wallet/common/brave_wallet.mojom

```diff
@@ -34,11 +35,15 @@ enum ProviderError {
                                  // specified chain
   kUnknownChain = 4902,          // The provider doesn't know this chain
 
-  kParsingError = -32700,        //Parse error - Invalid JSON
-  kInvalidRequest = -32600,      // JSON is not a valid request object
-  kMethodNotFound = -32601,      // Method does not exist
-  kInvalidParams = -32602,       // Invalid method parameters
-  kInternalError = -32603,       // Internal JSON RPC error
+  // Pre-defined error codes specified in
+  // https://www.jsonrpc.org/specification#error_object
+  kParsingError = -32700,    // Parse error - Invalid JSON
```

Comment:
Maybe better to have a define that has these and then use the define twice.

Reply:
I initially defined a JsonRpcError with these common values and have it written as `kParsingError  = JsonRpcError.kParsingError`, but ios build failed like below.
```
18:24:09  gen/brave/ios/browser/api/brave_wallet/brave_wallet.mojom.objc.h:35:42: error: expected expression
18:24:09    BraveWalletProviderErrorParsingError = <mojom.generate.module.EnumValue object at 0x10dba3670>,
18:24:09                                           ^
18:24:09  gen/brave/ios/browser/api/brave_wallet/brave_wallet.mojom.objc.h:35:43: error: use of undeclared identifier 'mojom'
18:24:09    BraveWalletProviderErrorParsingError = <mojom.generate.module.EnumValue object at 0x10dba3670>,
```

Reply:
I've asked @kylehickinson if it is something we could support on iOS, I'll left this PR as is for now.

Reply:
Fixed with 7d00722b470d11a49cecd41f74008a50726d9fce

Reply:
Addressed in https://github.com/brave/brave-core/pull/11836/commits/7d00722b470d11a49cecd41f74008a50726d9fce